python_modules/dagster/dagster_tests/core_tests/resource_tests/test_resource_parameters.py

```diff
@@ -303,3 +304,40 @@ def the_asset(context, config: AnAssetConfig, foo: ResourceOutput[str]):
         .success
     )
     assert executed["the_asset"]
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="requires python3.9")
+def test_no_err_builtin_annotations():
```

Comment:
Shouldn't we run this test on all versions

Reply:
Builtin types cannot be used as generics in python < 3.9, e.g.

```python
Python 3.8.12 (default, Aug  5 2022, 12:15:35)
>>> list[dict[str, str]]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

So this test only makes sense in >=3.9

Reply:
Can you add "Builtin types cannot be used as generics in python < 3.9, e.g." as a comment then?